DCOS-38939: Branding / Add Empty Header Bar

juliangieseke opened the pull request:
**DESCRIPTION**

This PR prepares UI for the HeaderBar (1. commit) and adds the bar itself (2. commit)

---

**HOW TO REVIEW**

`npm install` because ui-kit was updated, then make sure everything works as expected. I changed styles for sidebar and content, so please verify that sidebar collapses, expands, sticks and unsticks correctly. also check different browsers,  windows sizes and some modals please :)

---

**KNOWN ISSUES**

Well, Headerbar only contains foobar text… 

---

**HOW TO ENABLE**

In `src/js/components/HeaderBar.tsx`:
```diff
 export default function HeaderBar() {
-  //remove this to activate component
-  if (arguments) {
-    return null;
-  }
-
   return (
```

DanielMSchmidt commented:
@juliangieseke Snapshot tests are failing, please take a look

DanielMSchmidt commented on src/js/components/HeaderBar.tsx:
```diff
@@ -0,0 +1,18 @@
+import * as React from "react";
+import { HeaderBar as UIHeaderBar } from "#PLUGINS/ui-kit-candidates/HeaderBar";
```
I am unsure about the ui-kit-candidates approach here. I have these problems:

- For me, it communicates that all other components will never be moved to the ui-kit
- We now have two positions to look for ui components. We had the same at my last company and it leed to a "starvation" of the right place, because every component was "just" moved to the easier to integrate place with a light polish.

juliangieseke replied:
you are correct and I discussed it with philipp, we need to be careful with this directory and have to maintain it properly. the Idea was that we move everything that sits in there within a short period of time to ui-kit (or not). 

Do you want me to change something or create issues regarding the move?

DanielMSchmidt replied:
Can you explain to me why we can not add it to the `ui-kit`?

juliangieseke replied:
@nLight oh - gotcha! I'll fix that.
@DanielMSchmidt we can, and we should, but I wanted to move forward and optimize later. (same thing as in the tables PR - I'll create a Ticket)

DanielMSchmidt commented on plugins/ui-kit-candidates/HeaderBar.tsx:
```diff
@@ -0,0 +1,19 @@
+export class HeaderBar extends React.PureComponent {
+  render() {
+    const { children } = this.props;
+    return <div style={headerBarStyles}>{children}</div>;
```
🎨 this is one of the rare cases where I like `this.props.children` better 😆

juliangieseke replied:
agree 😆 there was more code before which I managed to strip out - that last refactoring slipped. I'll change it if I touch the file again.

juliangieseke commented:
@DanielMSchmidt  yup, ui kit updated changes classnames - pushing a fix as soon as linting passes 👍

DanielMSchmidt commented on plugins/ui-kit-candidates/HeaderBar.tsx:
```diff
@@ -0,0 +1,19 @@
+  flexShrink: 1,
+  flexGrow: 0,
+  flexBasis: "auto",
+  zIndex: 999
```
Would it make sense to put this into a constant, so we have named levels?

juliangieseke replied:
We could introduce something like that in UI Kit, yeah. (not in scope for this PR I'd say)

DanielMSchmidt replied:
Hmm, having UI Kit here is a problem. Ok nevermind

juliangieseke replied:
I dont think it is a problem actually, UI Kit can efine levels for certain types of components.

DanielMSchmidt commented on src/js/pages/Index.js:
```diff
@@ -150,15 +151,32 @@ var Index = React.createClass({
-        <ServerErrorModal />
+      <div
+        className={classSet}
+        style={{
```
Why do we need to use inline styles here?

juliangieseke replied:
because I had no idea how to deal with CNVS. styling there is such a mess and as we want to move to emotion with colocated styles anyway, I felt that having it as inline styles isnt _that_ bad. 🤷‍♂️ 

is that something you want me to change? or just a question?

nLight replied:
if it's a candidate for the ui-kit, shall we extract the styles the ui-kit way?

DanielMSchmidt replied:
Isn't this the index page? Why is it a ui-kit candidate?

DanielMSchmidt replied:
I would really like the styles being moved to CSS

nLight replied:
Please extract inline styles to `.application-wrapper`.

nLight commented:
What about the background color? I thought it's gonna be 💜 not black.

juliangieseke commented:
After discussion with @DanielMSchmidt and @Fabs , I changed `ui-kit-candidates` to `ui-kit-stage` and created a ticket moving the component to UI Kit 👍 (Also I added a small Readme to the folder)
Also addressed the comments /cc @nLight

nLight commented:
@juliangieseke there was no `npm install` step described 🤷‍♂️

nLight commented on src/js/components/Sidebar.js:
```diff
@@ -351,7 +351,10 @@ class Sidebar extends React.Component {
-        <div className="sidebar flex flex-direction-top-to-bottom">
+        <div
+          className="sidebar flex flex-direction-top-to-bottom"
+          style={{ position: "relative" }}
```
Can't we just add `position: relative` to the `sidebar` class?

nLight commented on src/js/pages/Index.js:
```diff
@@ -150,15 +151,32 @@ var Index = React.createClass({
+      >
+        <HeaderBar />
+        <div
+          style={{
```
👮 no inline styles

nLight commented on packages/ui-kit-stage/HeaderBar.tsx:
```diff
@@ -0,0 +1,23 @@
+import * as React from "react";
+import { coreColors } from "@dcos/ui-kit/dist/packages/shared/styles/color";
+
+const headerBarStyles: object = {
```
if it's a UI Kit candidate why not using emotion here? Example: https://github.com/dcos-labs/ui-kit/blob/master/packages/badge/components/badge.tsx

juliangieseke replied:
IMHO that would be part of the ui-kit issue, there is no emotion in dcos-ui yet 🤷‍♂️ If I finalize the component here, then there is no need for this staging folder thing… the Idea was to move on fast and optimize later instead of blocking this whole PR until that component is finalized 😕

weblancaster replied:
@juliangieseke @nLight I would not suggest adding this in the UI Kit. This case is more structural than functional/composable.

it's just returning a div with styles. what should be happening is the components inside this div are part of the UI Kit, not the div wrapper part of the UI Kit.

Fabs commented:
I took a swing at running it, but I will wait the next round of changes so I check the code.

juliangieseke commented:
@nLight @DanielMSchmidt moved styles into css files 👍

juliangieseke commented:
@Fabs ping :)
